tmt/utils.py

```diff
@@ -887,10 +1017,10 @@ def yaml_to_dict(data, check_version=False, yaml_type=None):
         yaml = YAML()
         result = yaml.load(data)
         return old_result == result, old_result
-    return old_result
+    return True, old_result
```

Comment:
Instead of always returning a tuple (which affects all occurences except for the only one used in the `lint` command) a thus makes the usage awkward, what about rather always returning just the converted `dict` and handling the `lint` check in on of the following ways:

 * raising an exception when the yaml 1.1 version is detected
 * comparing the outcome with 1.1 version directly under `lint` (or add and extra function)
 * completely forget about the 1.1 check, we've been warning for a long time :)

What do you think, @FrNecas?

Reply:
Hm, so I'm wondering where this check actually has an effect. The data that is used for execution is taken from fmf anyway which has been converted to `ruamel.yaml` so that should be 1.2 right? If we are only checking the 1.1 diff in linting on tmt side but `tmt run` (using fmf) results in something a bit different anyway, I think we can just drop this altogether? I was planning on dropping it in 2.0 (as stated in the comment above) but it's been a long time, there are only minor differences so I would hope that by now, we won't break anything for the users.

Reply:
Good point that we already consume `fmf` processed data. Agree to drop the extra check altogether.

Reply:
Revert to the old return value should be covered by 2839eb2. @happz, @FrNecas, please, have a look if I haven't missed anything.